src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,142 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value, null, null);
+        if (size == 0) {
+            first = newNode;
+        } else {
+            newNode.prev = last;
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> newNode;
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (index == 0) {
+            Node<T> newFirst = new Node<>(value, null, first);
+            first.prev = newFirst;
+            first = newFirst;
+        } else {
+            Node<T> node = findNode(index);
+            Node<T> prevNode = node.prev;
+            newNode = new Node<>(value, prevNode, node);
+            node.prev = newNode;
+            prevNode.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return findNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> oldNode = findNode(index);
+        T oldValue = oldNode.value;
+        oldNode.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> removedNode = findNode(index);
+        unLink(removedNode);
+        return removedNode.value;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> currentNode = first;
+        while (currentNode != null) {
+            if (elementsAreEqual(currentNode.value, object)) {
+                unLink(currentNode);
+                return true;
+            }
+            currentNode = currentNode.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
+        return size == 0;
+    }
+
+    private void checkIndex(int index) {
```

Comment:
method ordering. move this method after the place where it called first time - the method `findeNode()`